ZConfig/components/logger/loghandler.py

```diff
@@ -113,7 +114,10 @@ def rotate(self, rotateFilename=None):
         except OSError:
             pass
 
-        self.stream = open(self.baseFilename, self.mode)
+        if self.delay:
+            self.stream = None
+        else:
+            self.stream = open(self.baseFilename, self.mode, self.encoding)
```

Comment:
I don't think the call to `open` is correct here for either Python 2 or 3. On [Python 2](https://docs.python.org/2.7/library/functions.html#open) the signature is `open(name[, mode[, buffering]])`. On [Python 3](https://docs.python.org/3.7/library/functions.html#open) the signature is `open(file, mode='r', buffering=-1, encoding=None, errors=None, newline=None, closefd=True, opener=None)`. So both versions are going to interpret `self.encoding` as `buffering`. 

To get an `open` that supports `encoding=` on Python 2, you can use [io.open](https://docs.python.org/2.7/library/io.html#io.open)

I believe this code is [not tested by Travis](https://coveralls.io/builds/20434054/source?filename=ZConfig/components/logger/loghandler.py#L101) and is [excluded from the coverage report](https://github.com/zopefoundation/ZConfig/blob/master/.coveragerc#L16)

Reply:
You're right!  Fixed in 63465a8.